simpleble/src/backends/macos/PeripheralBaseMacOS.mm

```diff
@@ -19,6 +19,23 @@
         }                                         \
     } while (0)
 
+#define WAIT_UNTIL_FALSE_WITH_TIMEOUT(obj, var, timeout)                                      \
+    do {                                                                                      \
+        BOOL _tmpVar = YES;                                                                   \
+        auto _timeout = std::chrono::duration<double>(timeout);                               \
```

Comment:
We should not be using C++ constructs in here unless necessary. Also doubles are a very inefficient approach for this.

A variant of the old timeout logic makes more sense:
```
        NSDate* endDate = nil;

        // Wait for the connection to be established for up to 5 seconds.
        endDate = [NSDate dateWithTimeInterval:5.0 sinceDate:NSDate.now];
        while (self.peripheral.state == CBPeripheralStateConnecting && [NSDate.now compare:endDate] == NSOrderedAscending) {
            [NSThread sleepForTimeInterval:0.01];
        }
```